presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -550,7 +555,12 @@ else if (redistributeWrites) {
                 }
             }
 
-            if (partitioningScheme.isPresent() && !source.getProperties().isCompatibleTablePartitioningWith(partitioningScheme.get().getPartitioning(), false, metadata, session)) {
+            if (partitioningScheme.isPresent() &&
+                    // TODO: Deprecate compatible table partitioning
+                    !source.getProperties().isCompatibleTablePartitioningWith(partitioningScheme.get().getPartitioning(), false, metadata, session) &&
+                    !(partialMergePushdownStrategy == PUSHDOWN_BUT_NOT_THROUGH_JOIN_AND_AGGREGATION &&
+                            source.getProperties().isRefinedPartitioningOver(partitioningScheme.get().getPartitioning(), false, metadata, session) &&
+                            canPushdownPartialMerge(source.getNode()))) {
```

Comment:
Just to make sure I understand it correctly

We only optimizing simple projections/filters of a bucketed table into the other bucketed table? It doesn't save any memory, but potentially saves CPU cycles on doing shuffle.

Reply:
@arhimondr  Correct. For now it's mainly for completeness with some limited usage (read and directly write to a bucketed table with less number of buckets.

In the future, thinking about we can have more aggressive `ALWAY_PUSHDOWN_PARTIAL_MERGE` as `PartialMergePushdownStrategy` -- it might be useful in some use case (at the cost of more CPU...)